nestkernel/conn_builder.cpp

```diff
@@ -511,39 +500,22 @@ nest::ConnBuilder::set_synapse_params( DictionaryDatum syn_defaults, DictionaryD
 void
 nest::ConnBuilder::set_structural_plasticity_parameters( std::vector< DictionaryDatum > syn_specs )
 {
-  // Check if both pre and postsynaptic element are provided. Currently only possible to have
-  // structural plasticity with single element syn_spec.
-  bool have_both_sp_keys = false;
-  bool have_one_sp_key = false;
-  for ( auto syn_params : syn_specs )
+  DictionaryDatum syn_spec = syn_specs[ 0 ];
```

Comment:
Very nice cleanup here! If I read the code right, then if neither the pre- nor the postsynaptic element are given, the method will just do nothing. Is this intended behavior?

Reply:
Yes, it is.

Actually, I only inferred the functionality from the old code and was also confused initially. To avoid such confusion in the future, I have further restructured the code to be more like the original and added a docstring in 4ba134e2b7.